### PR TITLE
enhance conda post-link and pre-unlink scripts

### PR DESCRIPTION
**Pull request by jjhelmus**
Explicitly Pass "--prefix", the default should be fine but it is safer to be
explicit.

Simplify the scripts by removing the block, they are un-needed for single line
commands.

In the windows scripts set the PATH explicitly as most versions of conda do not
set this prior to calling the script. See:
https://github.com/conda/conda/issues/7789

### CODE REVIEW

**Comment by jjhelmus** [comment]
Once this is acceptable I'll rebuild the 2.2.0 conda packages to include this change as a short term fix for AnacondaRecipes/nb_conda_kernels-feedstock#1.

**Comment by mcg1969** [comment]
It looks like `conda` has been modified to disallow spaces in environment names. Unfortunately, I put a fair amount of effort into making sure that `nb_conda_kernels` handled such things properly.... it will take some effort to remove those tests.

File: conda-recipe/pre-unlink.bat [hunk 1]
```diff
@@ -1,4 +1,4 @@
 @echo off
-(
-  "%PREFIX%\python.exe" -m nb_conda_kernels.install --disable
-) >>"%PREFIX%\.messages.txt" 2>&1
+:: Set PATH explicitly as it may not be set correctly by some versions of conda
```
Comment (mcg1969) [inline conda-recipe/pre-unlink.bat]: I need to understand this a little better, @jjhelmus. Your comment seems to make it clear that the problem is with conda, not with this package. Why not just assume a correctly functioning conda?

One reason I am concerned about this is that I have found that conda's path manipulation scripts don't necessarily do a great job of handling things right when there is a combination of manual path manipulation _and_ conda's own activation mechanism. For instance, conda build effectively "double-activates" the test environment, and that throws conda off if you're trying to do activations as part of your tests (which will be happening here).

Comment (jjhelmus) [reply conda-recipe/pre-unlink.bat]: The comment on "some versions of conda" is forward thinking.  Currently ALL versions of conda do not set PATH when calling post link scripts (see conda/conda#7789).  The reason that this did not cause issues earlier is that on Windows the Python interpreter was patched to add "%PREFIX%\Library\bin" to path.  This patch was dropped from Anaconda's builds of Python 3.7.  The net result is that the post install script will fail most cases when `nb_conda_kernels` is included in a Python 3.7 environment.  More details can be found in [this comment](https://github.com/AnacondaRecipes/nb_conda_kernels-feedstock/issues/1#issuecomment-459873652)

Future versions should fix this bug, see conda/conda#8229

**Comment by dhirschfeld** [comment]
NB: I *think* this is the same underlying issue as https://github.com/AnacondaRecipes/qgrid-feedstock/pull/1

**Comment by jjhelmus** [comment]
> Why is this "safer"? If `nb_conda_kernels` is going to run properly, it's going to install itself into `sys.prefix`. Why explicitly specify something that is reliably recovered from the Python executable itself?

Safer was a bad choice of words.  My thinking in adding `--prefix` was to protect against cases where the incorrect python was run, but the post-link script uses $PREFIX%/python.exe so I can't imagine any cases where this would occur.  I'll revert this change.

File: conda-recipe/pre-unlink.bat (hunk 1 above)
Comment (mcg1969) [reply conda-recipe/pre-unlink.bat]: OK, I understand now and support this change.

**Comment by mcg1969** [comment]
@jjhelmus this is in master now. I'll be pushing a CHANGELOG update and tagging 2.2.1 soon.

**Pull request merged**